### PR TITLE
#51 obtain the leaf index of gbdt tree

### PR DESCRIPTION
**Pull request by grasevski**

### CODE REVIEW

File: lgensemble.go [hunk 1]
```diff
@@ -52,6 +52,13 @@ func (e *lgEnsemble) predictInner(fvals []float64, nEstimators int, predictions
 	}
 }
 
+func (e *lgEnsemble) predictLeaves(fvals []float64, predictions []int) error {
```
Comment (dmitryikh) [inline lgensemble.go]: seems like we don't need to return `error` here

File: xgensemble.go [hunk 2]
```diff
@@ -57,6 +57,13 @@ func (e *xgEnsemble) predictInner(fvals []float64, nEstimators int, predictions
 	}
 }
 
+func (e *xgEnsemble) predictLeaves(fvals []float64, predictions []int) error {
```
Comment (dmitryikh) [inline xgensemble.go]: we don't need to return error here?

**Comment by grasevski** [comment]
OK I've addressed your comments and only implemented for lightgbm as it seems the leaves aren't reordered in that implementation from what I can see. Also I'm only using lightgbm so that is ok for my use case.

**Comment by dmitryikh** [comment]
done in #76.

**Pull request closed without merging**